components/scream/src/physics/p3/p3_main_impl.hpp

```diff
@@ -185,10 +185,11 @@ ::p3_main_part1(
       // prescribe that value
 
       if (do_prescribed_CCN) {
```

Comment:
I was trying to argue this morning that this entire weird "we're in microphysics but we're going to do aerosol activation now" chunk of code below should be deleted in C++. If SPA does aerosol, we should expect the state after SPA has the correct nc value.

Reply:
Yeah, that would probably simplify things a lot.  For example we could set nccn_prescribed to a constant value in the interface if we are not using SPA.  I'm guessing we don't plan on predictingNC right?

Reply:
@PeterCaldwell , can you comment further on if you want me to take care of purging `do_prescribed_CCN` conditionals in this PR?